src/ui/scroll-shadower.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { FC, useEffect, useRef, useState } from 'react'
+import styled, { css } from 'styled-components'
+
+interface Props {
+  height: string
+  width: string
+}
+
+interface HorizontalShadowProps {
+  showOnLeft: boolean
+  showOnRight: boolean
+  rightOffset?: number
+}
+
+interface VerticalShadowProps {
+  showOnTop: boolean
+  showOnBottom: boolean
+  bottomOffset?: number
+}
+
+const ShadowHorizontal = styled.div<HorizontalShadowProps>`
+  position: relative;
+  ${({ showOnRight, rightOffset = 0 }) =>
+    showOnRight
+      ? css`
+          ::after {
+            content: '';
+            position: absolute;
+            top: 0;
+            bottom: 0;
+            right: ${rightOffset}px;
+            width: 30px;
+            background: linear-gradient(
+              to right,
+              rgba(255, 255, 255, 0),
+              rgba(0, 0, 0, 0.25)
+            );
+          }
+        `
+      : ''}
+  ${({ showOnLeft }) =>
+    showOnLeft
+      ? css`
+          ::before {
+            content: '';
+            position: absolute;
+            bottom: 0;
+            left: 0;
+            top: 0;
+            width: 30px;
+            background: linear-gradient(
+              to left,
+              rgba(255, 255, 255, 0),
+              rgba(0, 0, 0, 0.25)
+            );
+          }
+        `
+      : ''}
+`
+
+const ShadowVertical = styled.div<VerticalShadowProps>`
+  position: relative;
+  ${({ showOnTop }) =>
+    showOnTop
+      ? css`
+          ::before {
+            content: '';
+            position: absolute;
+            top: 0;
+            left: 0;
+            right: 0;
+            height: 30px;
+            background: linear-gradient(
+              to top,
+              rgba(255, 255, 255, 0),
+              rgba(0, 0, 0, 0.25)
+            );
+          }
+        `
+      : ''}
+  ${({ showOnBottom, bottomOffset = 0 }) =>
+    showOnBottom
+      ? css`
+          ::after {
+            content: '';
+            position: absolute;
+            bottom: ${bottomOffset}px;
+            left: 0;
+            right: 0;
+            height: 30px;
+            background: linear-gradient(
+              to bottom,
+              rgba(255, 255, 255, 0),
+              rgba(0, 0, 0, 0.25)
+            );
+          }
+        `
+      : ''}
+`
+
+const Container = styled.div<Props>`
+  height: ${({ height }) => height};
+  width: ${({ width }) => width};
+  overflow-x: auto;
+  display: flex;
+`
+
+export const ScrollShadower: FC<Props> = ({ children, ...props }) => {
+  const ref = useRef<HTMLDivElement>(null)
+
+  const [showOnLeft, setShowOnLeft] = useState<boolean>(false)
```

Comment:
Please, don't use `useState` for update visibility of shadows. Also, save offset in state is very bad for performance.

You just need to change `display` css property direct for your shadow elements.

And you can create this shadow elements not in JSX, just `document.createElement('div')` in your setup effect.